packages/integration-sdk-entity-validator/src/validator.ts

```diff
@@ -0,0 +1,69 @@
+import Ajv, { AnySchema, ErrorObject } from 'ajv';
+import addFormats from 'ajv-formats';
+import { addJ1Formats } from './j1Formats';
+import { ajvErrorToEntityValidationError } from './entityValidationError';
+
+export class EntityValidator {
+  private ajvInstance: Ajv;
+
+  constructor({ schemas }: { schemas?: AnySchema[] }) {
+    this.ajvInstance = addJ1Formats(
+      addFormats(
+        new Ajv({
+          strictSchema: false,
+          allErrors: true,
+        }),
+      ),
+    );
+
+    if (schemas) {
+      this.addSchemas(schemas);
+    }
+  }
+
+  addSchemas(schema: AnySchema | AnySchema[]) {
+    this.ajvInstance.addSchema(schema);
+    return this;
+  }
+
+  validateEntity<Entity extends { _class: string[] | string; _type: string }>(
+    entity: Entity,
+  ) {
+    const classArray = Array.isArray(entity._class)
+      ? entity._class
+      : [entity._class];
+    let errors: ErrorObject[] = [];
+
+    const typeValidator = this.ajvInstance.getSchema(`#${entity._type}`);
+
+    if (typeValidator) {
+      const isValid = typeValidator(entity);
+
+      if (!isValid) {
+        errors = [...errors, ...(typeValidator.errors ?? [])];
+      }
+    } else {
+      for (const className of classArray) {
+        const validator = this.ajvInstance.getSchema(`#${className}`);
+
+        if (!validator) {
+          throw new Error(`Schema not found for class "${className}"`);
+        }
```

Comment:
...which is why I don't like this. Throwing errors from inside a for loop is too "aggressive".
Can't we also have this information returned?
If we don't find a _class it makes sense to me that it would just be skipped, instead of halting the entire validation.

Reply:
That a really good point. I'm going to return a list of validated and not validated classes and let the consumer handle if it should throw or not the exception.